doc/user/content/integrations/terraform.md

```diff
@@ -0,0 +1,118 @@
+---
+title: "Terraform Materialize Provider"
+description: "Create and manage Materialize objects with the Terraform Materialize provider"
+menu:
+  main:
+    parent: "integrations"
+    weight: 11
+    name: "Terraform Materialize Provider"
+---
+
+[Terraform](https://www.terraform.io/) is an infrastructure-as-code tool that allows you to manage your
+resources in a declarative configuration language. Materialize maintains a
+Terraform provider to help you configure Materialize resources like
```

Comment:
Related to the thread below that @sjwiesman started, do we want this page to focus just on the provider (materialize objectives) or also the additional modules (external objects)? For now, I'd be in favor of the latter, but we'd have to get clearer about this in the intro and throughout.